### ISSUE TITLE
Does SU2 code have capability of getting and solving linearized NS in matrix form?

### ISSUE THREAD

**Issue by monika1387** [su2code/SU2#1141]
**Is your feature request related to a problem? Please describe.**
*A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]*

**Describe the solution you'd like**
*A clear and concise description of what you want to happen.*

**Describe alternatives you've considered**
*A clear and concise description of any alternative solutions or features you've considered.*

**Additional context**
*Add any other context or screenshots about the feature request here.*

**Comment by pcarruscag** [comment]
Please add a little more information on what you would like, if this is just a question, I think the answer is no.

**Comment by monika1387** [comment]
We are trying to perform resolvent analysis and need to linear NS equations in matrix form. But I guess I got your answer. Thank you for help!

**Comment by pcarruscag** [comment]
That is not more information, just the tittle again. If you meet me halfway and explain me in layman's terms what you need the code to do I can give advise on what parts to modify and how, as I have done in the past for other people.
Otherwise I can tell you that in 3 years I never saw a feature_request being fulfilled.

**Comment by monika1387** [comment]
Hi @pcarruscag  Let me give me some background, we are trying to perform resolvent analysis and for that we need to have linearized NS equations. I ran my LES simulations in SU2 and mean flow data is available in SU2 so I would like to stick to SU2 to understand if SU2 can solve the “direct" linearized Navier-Stokes equations? 
Also, we are looking for PETSc library to use for LU decomposition. I understand SU2 uses eigen numerical library so possibly we can integrate PETSc in SU2 or perhaps we can perform outside SU2 after above step is completed to get the solved linearized matrices in SU2.  I tried to explain please let me know if this make sense, else I will provide more details. 
I am referencing this paper for my work page 4-6 (https://arxiv.org/abs/2005.10964)
[Optimal eddy viscosity for resolvent-based.pdf](https://github.com/su2code/SU2/files/5765838/Optimal.eddy.viscosity.for.resolvent-based.pdf)

**Comment by pcarruscag** [comment]
On the LU part we have an interface with PaStiX.
In SU2 Eigen is not used to solve distributed(mpi)-sparse systems, just small dense problems. Also to my knowledge Eigen does not support MPI at least a couple years ago its interface with PaStiX was threads only.
I opted to develop the interface with PaStiX rather than PETSc because it seemed the support for the block-sparse format we use was limited, i.e. compatible with only a few linear solvers.
There are instructions on how to build SU2 with PaStiX in the TestCases folder.

I had a quick look at your paper, in SU2 the linearization of F (the A matrix you need, I believe) is not exact, for an exact linearization you would need to use forward AD but then any system would need to be solved iteratively in a matrix-free products.

**Comment by monika1387** [comment]
@pcarruscag  Thank you this is really helpful lead in PaStiX. I will look more into this. Appreciate your comments!
